src/util/ast.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Node as AstNodeInternal,
+  JSONPath,
+  findNodeAtLocation as findNodeAtLocationInternal,
+  parseTree as parseTreeInternal,
+  ParseError,
+} from 'jsonc-parser';
+import {PlaceholderConfig} from '../analyzer.js';
+import {WireitError} from '../error.js';
+export {ParseError} from 'jsonc-parser';
+
+type ValueTypes = string | number | boolean | null | undefined;
+
+/**
+ * A JSON AST node.
+ *
+ * A safer override, preferring unknown over any.
+ */
+export interface AstNode<T extends ValueTypes = ValueTypes>
```

Comment:
Consider `JsonAstNode` or even `JsonNode`? I sort of feel that knowing this is connected to JSON is the more useful thing to know.

Reply:
Sure. My thinking here was just that there's only one kind of AST that we're dealing with, but JsonAstNode is less ambiguous